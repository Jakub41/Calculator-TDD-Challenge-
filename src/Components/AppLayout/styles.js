import styled from "@emotion/styled";

export const ContentLayout = styled.div`
	width: 100px;
	height: 100px;
`
