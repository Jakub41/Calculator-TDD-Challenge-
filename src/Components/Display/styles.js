import styled from "@emotion/styled";

export const DisplayDiv = styled.div`

`;
