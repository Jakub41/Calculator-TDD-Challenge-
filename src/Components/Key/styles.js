import styled from "@emotion/styled";

export const KeyDiv = styled.div`

`;

export const KeyButton = styled.button`

`;

