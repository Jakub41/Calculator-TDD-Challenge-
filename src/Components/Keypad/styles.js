import styled from "@emotion/styled";

export const KeypadDiv = styled.div`

`;

export const NumbersDiv = styled.div`

`;

export const OperatorsDiv = styled.div`

`;
