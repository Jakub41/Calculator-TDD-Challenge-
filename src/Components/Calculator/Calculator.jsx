import React, { useState } from "react";
import { CalculatorDiv } from "./styles";

const Calculator = () => {
	return (
		<>
			<CalculatorDiv></CalculatorDiv>
		</>
	);
};

export default Calculator;
