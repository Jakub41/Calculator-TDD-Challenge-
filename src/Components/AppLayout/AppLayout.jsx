import React from "react";
import { ContentLayout } from "./styles";
import Calculator from "../Calculator";

const AppLayout = () => {
	return (
		<>
			<ContentLayout>
				<Calculator />
			</ContentLayout>
		</>
	);
};

export default AppLayout;
