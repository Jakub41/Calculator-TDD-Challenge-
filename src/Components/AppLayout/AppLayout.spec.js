import { shallow } from "enzyme";
import React from "react";
import renderer from "react-test-renderer";
import AppLayout from "./AppLayout";
import { ContentLayout } from "./styles";
import Calculator from "../Calculator/Calculator";

// tests if we render the layout correctly with the styled component
describe("AppLayout Component", () => {
	let wrapper;
	beforeEach(() => (wrapper = shallow(<AppLayout />)));

	it("AppLayout Snapshot renderer", () => {
		const component = renderer.create(<AppLayout />);
		let tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("ContentLayout Snapshot renderer", () => {
		const styledComponent = renderer.create(<ContentLayout />);
		let tree = styledComponent.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("Renders component with styled component ContentLayout", () => {
		expect(wrapper.find(ContentLayout)).toHaveLength(1);
	});

	// test if CalculatorComponent is rendered under AppLayout Component
	it("should render the Calculator Component", () => {
		expect(wrapper.containsMatchingElement(<Calculator />)).toEqual(true);
	});
});
