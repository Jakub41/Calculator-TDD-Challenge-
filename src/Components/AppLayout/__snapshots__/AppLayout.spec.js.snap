// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayout Component AppLayout Snapshot renderer 1`] = `
<div
  className="css-14fwp1f"
>
  <div
    className="css-1esthja"
  />
</div>
`;

exports[`AppLayout Component ContentLayout Snapshot renderer 1`] = `
<div
  className="css-14fwp1f"
/>
`;
